x-pack/platform/plugins/shared/task_manager/server/task_claimers/strategy_mget.ts

```diff
@@ -91,10 +91,19 @@ async function claimAvailableTasks(opts: TaskClaimerOpts): Promise<ClaimOwnershi
   const initialCapacity = getCapacity();
   const stopTaskTimer = startTaskTimer();
 
+  const taskTypes = new Set<string>();
```

Comment:
I think the problem lies in https://github.com/elastic/kibana/blob/2e6f75380fee2132f99604211fc5a002ee90fbd2/x-pack/platform/plugins/shared/task_manager/server/task_pool/cost_capacity.ts#L72

where we're just checking for `taskDefinition.maxConcurrency`, so not handling the max concurrency of 0 correctly. we're already getting the limited and unlimited batches in the `buildClaimPartitions` function below but the capacity is returning incorrectly when the maxConcurrency is 0

Reply:
I didn't even check that part because `update_by_query` relies on the batches to filter out the skipped tasks, just applied the same logic here.
Thanks for pointing the above problem, moved the fix there.
I guess the same fix should be applied to `worker_capacity` as well. Fixed that too.